### PR TITLE
Screen GEFS grid2obs Aeronet file before ASCII2NC call Feature/aeronetasciifix

### PR DESCRIPTION
**Pull request by Ho-ChunHuang-NOAA**
<b>Note to developers: You must use this PR template!</b>

## Description of Changes

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

To avoid ERROR of corrupted input obs file, this code will screen input aeronet file for broken record (inconsistent columns per record) and write out record with consistent column number. The checked aeronet file will be used in the ASCII2NC conversion.  Minor fix for sending email in the stats step for reduced GEFS grib2 files that can be corrupted (zero size) due to disk quota exceed.  This is for emc parallel to spot something wrong in the prep step.

## Developer Questions and Checklist
* Is this a high priorty PR? If so, why and is there a date it needs to be merged by?
NO

* Do you have any planned upcoming annual leave/PTO?
NO

* Are there any changes needed for when the jobs are supposed to run?
NO
  
- [ ] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
YES

- [ ] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
YES

- [ ] References the feature branch for `HOMEevs` are removed from the code.
YES

- [ ] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
YES

- [ ] Jobs over 15 minutes in runtime have restart capability.
Jobs does not exceed 15 mins.

- [ ] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
N/A

- [ ] Jobs contain the appropriate file checking and don't run METplus for any missing data.
YES

- [ ] Code is using METplus wrappers structure and not calling MET executables directly.
YES

- [ ] Log is free of any ERRORs or WARNINGs.
YES

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.

For prep step ONLY

The corrupted aeroent file existed on VDATE=20241014 and in personal archived directory.

Add 
export DCOMINaeronet=/lfs/h2/emc/vpppg/noscrub/ho-chun.huang/dcom/prod/aeronet

in 

/dev/drivers/scripts/prep/global_ens/jevs_global_ens_chem_grid2obs_prep.sh for TESTING ONLY

One should see message
"DEBUG :: Line 4317 has different columns number 137 vs standard 113"

and no ERROR and WARNING message in the log

### CODE REVIEW

File: parm/metplus_config/prep/global_ens/chem_grid2obs/ASCII2NC_obsAERONET.conf
Comment (malloryprow) [inline parm/metplus_config/prep/global_ens/chem_grid2obs/ASCII2NC_obsAERONET.conf]: Remove commented out line

Comment (Ho-ChunHuang-NOAA) [reply parm/metplus_config/prep/global_ens/chem_grid2obs/ASCII2NC_obsAERONET.conf]: done

**Comment by malloryprow** [comment]
### jevs_global_ens_chem_grid2obs_prep.sh - Bad Data

Ran with INITDATE=20241014 and DCOMINaeronet=/lfs/h2/emc/vpppg/noscrub/ho-chun.huang/dcom/prod/aeronet

**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr593/EVS/dev/drivers/scripts/prep/global_ens/jevs_global_ens_chem_grid2obs_prep.o158973348
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr593/evs/v2.0
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_global_ens_chem_grid2obs_prep.158973348.cbqs01

**Comment by Ho-ChunHuang-NOAA** [comment]
The log file looks good to me. 

The /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr593/evs/v2.0/prep/global_ens/chem.20241014/gefs/aeronet_All_20241014_lev15.nc looks good as well

**Comment by Ho-ChunHuang-NOAA** [comment]
Wait for me to find out why there is a file "6" produced in the DATA.

**Comment by malloryprow** [comment]
Do we have a good date we can test to make sure all is okay there too? If I remove the INITDATE and DCOMINaeronet, will testing today's run be okay?

**Comment by malloryprow** [comment]
Waiting!

**Comment by Ho-ChunHuang-NOAA** [comment]
Find the problem, but it won't impact PR testing already done.

Please go ahead to run today's prep after the pull.

**Comment by malloryprow** [comment]
### jevs_global_ens_chem_grid2obs_prep.sh - Good Data

Ran with INITDATE=20241019 and DCOMINaeronet=/lfs/h1/ops/prod/dcom

**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr593/EVS/dev/drivers/scripts/prep/global_ens/jevs_global_ens_chem_grid2obs_prep.o158974709
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr593/evs/v2.0
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_global_ens_chem_grid2obs_prep.158974709.cbqs01

**Comment by Ho-ChunHuang-NOAA** [comment]
Today's prep is successful.

**Review by malloryprow — APPROVED** [review]
Code changes are good and testing successful.

**Comment by malloryprow** [comment]
@AndrewBenjamin-NOAA This needs your review (and pending approval) when you get a chance.

**Review by AndrewBenjamin-NOAA — APPROVED** [review]
I have reviewed the code and accept these changes. Thank you for adding additional checks to the aeronet files.

**Comment by malloryprow** [comment]
@Ho-ChunHuang-NOAA Can you do a sync fork before I merge?

**Comment by Ho-ChunHuang-NOAA** [comment]
done

**Pull request merged**
